### DISCUSSION TITLE
code-file example not working

### DISCUSSION THREAD

**Discussion by simplecommerce** [graphql-hive/graphql-eslint#1104] [Q&A] [answered]
Hi,

I have followed the example for the code-file for my *.js files and I keep getting these 2 errors:

1:0  error  Parsing error: [graphql-eslint] Syntax Error: Unexpected Name "const"
1:0  error  Parsing error: [graphql-eslint] Syntax Error: Unexpected Name "export"

In my *.js files I have the following:

```
const { gql } = require('apollo-server');

export default gql`
# schema here
`
```

In my .eslintrc.json file I have the following config

```
{
    "root": true,
    // ❗️ It's very important that you don't have any rules configured at the top-level config,
    // and to move all configurations into the overrides section. Since JavaScript rules
    // can't run on GraphQL files and vice versa, if you have rules configured at the top level,
    // they will try to also execute for all overrides, as ESLint's configs cascade
    "overrides": [
      {
        "files": ["*.js"],
        "processor": "@graphql-eslint/graphql",
        "extends": ["eslint:recommended","plugin:@graphql-eslint/schema-recommended"],
        "plugins": ["@graphql-eslint"],
        "env": {
            "es2021": true,
            "node": true
        },
        "parserOptions": {
            "ecmaFeatures": {
              "ecmaVersion": "latest",        
              "globalReturn": false
            },
            "sourceType": "module"
        }        
    }
}
```

I don't understand why it is always stopping at the `const` and `export`.

Am I missing something?

Sorry if the answer is obvious.

**Comment by simplecommerce** [comment]
I figured my issue, sorry about that.

This is what I ended up with.

```
{
    "root": true,
    // ❗️ It's very important that you don't have any rules configured at the top-level config,
    // and to move all configurations into the overrides section. Since JavaScript rules
    // can't run on GraphQL files and vice versa, if you have rules configured at the top level,
    // they will try to also execute for all overrides, as ESLint's configs cascade
    "overrides": [
      {
        "files": ["*.js"],
        "processor": "@graphql-eslint/graphql",
        "extends": ["eslint:recommended","plugin:prettier/recommended"],
        "plugins": ["@graphql-eslint"],
        "env": {
            "es6": true,
            "node": true
        },
        "parserOptions": {
            "sourceType": "module"
        }
    }
]   
}
```

**Reply by simplecommerce** [reply]
Odd, this doesn't work and keeps giving me the error with `const`
```
{
    "root": true,
    // ❗️ It's very important that you don't have any rules configured at the top-level config,
    // and to move all configurations into the overrides section. Since JavaScript rules
    // can't run on GraphQL files and vice versa, if you have rules configured at the top level,
    // they will try to also execute for all overrides, as ESLint's configs cascade
    "overrides": [
      {
        "files": ["*.js"],
        "extends": ["eslint:recommended","plugin:prettier/recommended","plugin:@graphql-eslint/schema-recommended"],
        "env": {
            "es6": true,
            "node": true
        },
        "parserOptions": {
            "sourceType": "module"
        }
    }
]   
}
```

**Reply by simplecommerce** [reply]
Ok I think I know what my problem is, it is because I am not using `*.graphql` schema files.
And I can't seem to get it to work without that.
Does this plugin actually work without `*.graphql` files if I only have my schemas in `*.js` code files?

**Comment by dimaMachina** [comment, marked as answer]
Your configuration is invalid, follow this example https://github.com/B2o5T/graphql-eslint/blob/master/examples/monorepo/.eslintrc.cjs to found what’s wrong in your configuration

**Reply by simplecommerce** [reply]
@B2o5T thanks, but from what I am seeing in the example link, it keeps referring to `SCHEMA_PATH`, what if I don't have `*.graphql` files?

All my schemas are in code only `*.js`.

**Reply by dimaMachina** [reply]
Update `SCHEMA_PATH` to match your project (`<your-folder>/**/*.js`) and this line  https://github.com/B2o5T/graphql-eslint/blob/master/examples/monorepo/.eslintrc.cjs#L45 to `<your-folder>/**/*.js/*.graphql` or `<your-folder>/**/*.graphql`

**Reply by dimaMachina** [reply]
because `graphql-eslint` extract graphql documents under virtual filename that ends with `ORIGINAL_FILENAME/INDEX_document.graphql`, where `INDEX` is 0 for the first extracted document, 1 for the second, etc

**Reply by simplecommerce** [reply]
@B2o5T sorry about this, this is what I am trying and it doesn't seem to work, it keeps giving me this error:

>   1:0  error  Parsing error: [graphql-eslint] Syntax Error: Unexpected Name "const"

```
{
  "root": true,
  "overrides": [
    // I put this to indent my gql templates
    {
      "files": [
        "*.js"
      ],
      "extends": [
        "eslint:recommended",
        "plugin:prettier/recommended"
      ],
      "env": {
        "es2021": true,
        "node": true
      },
      "parserOptions": {
        "sourceType": "module"
      }
    },
    {
     // I put this because my schemas are all in code in *.js files
      "files": "*.js",
      "parser": "@graphql-eslint/eslint-plugin",
      "plugins": [
        "@graphql-eslint"
      ],
      "parserOptions": {
        // i put this because you told me to replace SCHEMA_PATH with the path to my *.js files
        "schema": "*.js"
      }
    },
    {
     // I put this because you told me to point it to *.js/*.graphql 
      "files": "*.js/*.graphql",
      "extends": "plugin:@graphql-eslint/schema-recommended",
      "rules": {
        // I put this rule to test it
        "@graphql-eslint/input-name": [
          "error",
          {
            "checkInputType": true
          }
        ]
      }
    }
  ]
}
```

I do not have static *.graphql or *.gql files, I do not have operations files either only schemas.
I am not able to see what I am doing wrong.

My schema file looks like this:

```
const { gql } = require("apollo-server");

export default gql`

`
```

**Reply by dimaMachina** [reply]
```
// I put this because my schemas are all in code in *.js files
      "files": "*.js",
```
I didn’t asked you to change this, leave this as `*.graphql`

**Reply by simplecommerce** [reply]
@B2o5T sorry about that, I forgot to mention I tried it before like this:

```
{
    "root": true,
    "overrides": [
      // I put this to indent my gql templates
      {
        "files": [
          "*.js"
        ],
        "extends": [
          "eslint:recommended",
          "plugin:prettier/recommended"
        ],
        "env": {
          "es2021": true,
          "node": true
        },
        "parserOptions": {
          "sourceType": "module"
        }
      },
      {
       // I put this because my schemas are all in code in *.js files
        "files": "*.graphql",
        "parser": "@graphql-eslint/eslint-plugin",
        "plugins": [
          "@graphql-eslint"
        ],
        "parserOptions": {
          // i put this because you told me to replace SCHEMA_PATH with the path to my *.js files
          "schema": "*.js"
        }
      },
      {
       // I put this because you told me to point it to *.js/*.graphql 
        "files": "*.js/*.graphql",
        "extends": "plugin:@graphql-eslint/schema-recommended",
        "rules": {
          // I put this rule to test it
          "@graphql-eslint/input-name": [
            "error",
            {
              "checkInputType": true
            }
          ]
        }
      }
    ]
  }
```

I did as you just said by leaving it to `*.graphql* and in my schema file I have done this as a test

```
const { gql } = require("apollo-server");

export default gql`
	type Mutation {
		refreshToken(input: RefreshToken!): 
	}
`	
```

When I run the lint, nothing happens, it does not show me an error saying the input type is wrong.

This is why I wasn't able to figure out what I was missing.

**Reply by dimaMachina** [reply]
Add a minimal reproduction repo and I’ll try to help you

**Reply by simplecommerce** [reply]
@B2o5T here is the repo: https://github.com/simplecommerce/graphql-eslint

I am using node v14.18.1 in case you need this info.
All I did was

`npm install` and then `npx eslint authentication.js` 

And nothing changes in the file.

Thanks for your help I appreciate it.

**Reply by dimaMachina** [reply]
@simplecommerce the issue was due missing `"processor": "@graphql-eslint/graphql",` in `*.js` override

Note: processor is needed to extract graphql document from code-files (`.js`/`.ts`)

**Reply by simplecommerce** [reply]
@B2o5T thanks for your help, I appreciate the time, I got it semi working and am getting issues but I think its just the way my project is coded/defined.

Another quick question, I assume this plugin doesn't support schema merging?

In my current project my schemas are defined in *.js files and some files can use schema types from other files, in my project main file, I merge my schemas into one single schema.

I assume this won't work.

**Reply by dimaMachina** [reply]
If you want to use `schema-recommended` you must provide a full schema for `graphql-eslint`. If you use some types from other project you can manually extend your schema as I described in this comment https://github.com/B2o5T/graphql-eslint/discussions/1063#discussioncomment-2721160
